src/test/kotlin/oop/Proxy/ScreenVirtualProxyTest.kt

```diff
@@ -8,9 +8,15 @@ class ScreenVirtualProxyTest {
 
   @Test
   fun `not create real object in proxy construction`() {
-    val screenVirtualProxy = ScreenVirtualProxy({ RealScreen() })
+    var createCount = 0
+    var createRealScreen = {
+      createCount++
+      RealScreen()
+    }
+
+    val screenVirtualProxy = ScreenVirtualProxy(createRealScreen)
 
-    assertNull(screenVirtualProxy.realScreen)
+    assertThat(createCount, `is`(0))
```

Comment:
This may be the solution we are looking for. We need to know yet if it is creating RealScreen for returning it as soon as it gets called or if he waits until he is called to create 🤔